### ISSUE TITLE
The use of attribute being defined outside of __init__ in derived class not being reported if the attribute is also defined in the base class.

### ISSUE THREAD

**Issue by asincero** [PyCQA/pylint#2622]
### Steps to reproduce
    class Base:
        def __init__(self) -> None:
            self.__foobar = 'foobar'

    class Derived(Base):
        def do_it(self) -> None:
            self.__foobar = 'Derived foobar'
            print(self.__foobar)

    o = Derived()
    o.do_it()

### Current behavior

pylint doesn't detect the assignment of "self.__foobar" in Derived.do_it() as an error.  If it was some other attribute, it would correctly report that an attribute was being defined on self outside of \_\_init\_\_.

### Expected behavior

pylint should report that "self.__foobar" was being defined outside of \_\_init\_\_ in Derived.do_it().

### pylint --version output

pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Sep 20 2018, 14:56:27) 
[GCC 6.3.0 20170516]

**Comment by PCManticore** [comment]
Thanks @asincero

**Comment by MarKett** [comment]
This ticket is open despite this seems to be the current behavior in the newest version.

But the more important question for me is: Why should this behavior be expected? If you have initialized the member in the base class, why and how should you initialize it again, without adding a otherwise useless __init__ in the derived class?

**Comment by Pierre-Sassoulas** [comment]
@MarKett the attribute is private in ``Base`` so it is not defined in ``Derived``. Consider the following:
```python
class Base:
    def __init__(self) -> None:
        self.__foobar = 'foobar'

class Derived(Base):

    def __init__(self):
        super(Derived, self).__init__()
        print(self.__foobar)


Derived()
```
Result of python a.py:
```
  File "a.py", line 12, in <module>
    Derived()
  File "a.py", line 9, in __init__
    print(self.__foobar)
AttributeError: 'Derived' object has no attribute '_Derived__foobar'

```

**Comment by Pierre-Sassoulas** [comment]
I'm going to close in favor of https://github.com/PyCQA/pylint/issues/2620
